Modules/Numerics/NarrowBand/include/itkNarrowBandImageFilterBase.h

```diff
@@ -292,7 +292,8 @@ class ITK_TEMPLATE_EXPORT NarrowBandImageFilterBase : public FiniteDifferenceIma
 
   bool m_Touched;
 
-  std::vector<bool> m_TouchedForThread;
+  // NB: although semantically boolean, int is used because vector<bool> is not thread safe due to the possibility of multiple bits being packed together in the same memory location.
+  std::vector<int> m_TouchedForThread;
```

Comment:
`m_ValidTimeStepList` and `m_TouchedForThread` are clearly two different cases, although they appear to have the same thread safety error, as you found out. I would rather deal with them in two separate commits.

In case of `m_TouchedForThread`, replacing `std::vector<bool>` by `std::vector<std::int8_t>` (or something similar)  would be just fine to me. While it seems to me that `m_ValidTimeStepList` may just be removed (including the corresponding `ResolveTimeStep` parameter). OK?